src/test/java/com/google/sps/MealServletTest.java

```diff
@@ -0,0 +1,233 @@
+
+        assertEquals(expected, actual);
+    }
+
+    // Get an object from empty datastore.
+    // Result: Response Status NOT FOUND.
+    @Test
+    public void getMealByIdFromEmptyDsTest() throws IOException, ServletException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        
+        MealServlet servlet = new MealServlet();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        request.setPathInfo("/2");
+        
+        try {
+            servlet.doGet(request, response);
+        } catch(Exception e) {
+            System.out.println(e);
+        }
+        MockHttpServletResponse expectedResponse = new MockHttpServletResponse();
+        expectedResponse.setStatus(HttpServletResponse.SC_NOT_FOUND);
+        int expected = expectedResponse.getStatus();
+        int actual = response.getStatus();
+
+        assertEquals(expected, actual);
+    }
+
+    // From datastore, where two entities with same id exist, get an object by this id.
+    // Result: Response Status INTERNAL SERVER ERROR.
+    @Test
+    public void getMealByIdForMultipleEntitiesTest() throws IOException, ServletException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        ds.put(createMealEntity(MEAL_1));
+        ds.put(createMealEntity(MEAL_1_DUPLICATE));
+        
+        MealServlet servlet = new MealServlet();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        request.setPathInfo("/1");
+        
+        try {
+            servlet.doGet(request, response);
+        } catch(Exception e) {
+            System.out.println(e);
+        }
+        MockHttpServletResponse expectedResponse = new MockHttpServletResponse();
+        expectedResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        int expected = expectedResponse.getStatus();
+        int actual = response.getStatus();
+
+        assertEquals(expected, actual);
+    }
+
+    // Get a Meal object from datastore, that can't be created (invalid field value).
+    // Result: catching ClassCastException().
+        @Test
+    public void getEmptyMealByIdTest() throws IOException, ServletException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        ds.put(createMealEntity(MEAL_EMPTY));
+        ds.put(createMealEntity(MEAL_1));
+        
+        MealServlet servlet = new MealServlet();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        request.setPathInfo("/0");
+        
+        try {
+            servlet.doGet(request, response);
+        } catch(ClassCastException e) {
+            System.out.println(e);
+            assertTrue(true);
+        }
```

Comment:
`doGet()` shouldn't throw an exception in this case - both because it's implemented that way *and* because it's good design to carry on even if *some* errors are in the way.

Result I expect: either `SC_NOT_FOUND` or `SC_INTERNAL_SERVER_ERROR`.

Also this test passes whether the exception is caught or not. If exception was thrown, it gets caught here and `assert` passes - it's ok. If the exception is not thrown, then it's not caught, so the `assert` line is never executed. Therefore 0 asserts out of 0 encountered are satisfied, the test passes too.

First and foremost: doGet() shouldn't throw here. Instead it should return a response with an error code.
...
...
But for the future - to actually test if something throws as expected, you need to use `assertThrows()`.